src/shared/dissect-image.c

```diff
@@ -2265,6 +2273,11 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
         BLOCK_SIGNALS(SIGCHLD);
 
         assert(m);
+        assert(m->name);
+
+        /* As per the os-release spec, if the image is an extension it will have a file
+         * named after the image name in extension-release.d/ */
+        paths[META_EXTENSION_RELEASE] = strjoina("/usr/lib/extension-release.d/extension-release.", m->name);
```

Comment:
hmm, i'd be more comfortable if we'd first check if `m->name` qualifies as image name... i.e. either by image_name_is_valid() or by hostname_is_valid()

i figure the latter. i.e. strip the version of the image name

Reply:
Change to image_name_is_valid, which is used elsewhere here as well